modules/govuk/templates/node/s_backend_lb/rate-limiting.conf.erb

```diff
@@ -26,3 +26,15 @@ limit_req_zone $post_limit_req zone=contact:5m rate=6r/m;
 # Return 429 (Too Many Requests) instead of the default 503.
 limit_req_status 429;
 limit_conn_status 429;
+
+# Create a rate limiting zone that handles limiting excessive load from
+# a specific high-load google apps script by ID. This may need to be changed to limit 
+# all Google-App-Script calls.
+map $http_user_agent $script_limit {
+  default "";
+  ~*(UAEmdDd-Tj3yuNDB-5kGWRZCbhvY_Im35Vdc) $binary_remote_addr;
```

Comment:
I don't think we should limit by this specific ID.  If a script with ID "foo" makes a million requests in a second, we should limit that too.

I'd instead like to see something which extracts the ID from the user agent and uses that.

Reply:
So you're thinking we store it somewhere? One option I do have is to limit it based on type e.g. `google-apps-script`

Reply:
I wasn't sure if that was the best approach.

Reply:
So, we currently have rate limiting based on IPs, right?  If the same IP sends too many requests in a short period of time, we serve a 429 response.

It would be good if we could do the same thing here, but based on the user-agent, where it's a google apps scripts user-agent.